presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcMetadata.java

```diff
@@ -82,17 +80,28 @@ public JdbcTableHandle getTableHandle(ConnectorSession session, SchemaTableName
     }
 
     @Override
-    public List<ConnectorTableLayoutResult> getTableLayouts(ConnectorSession session, ConnectorTableHandle table, Constraint constraint, Optional<Set<ColumnHandle>> desiredColumns)
+    public Optional<ConstraintApplicationResult<ConnectorTableHandle>> applyFilter(ConnectorTableHandle handle, Constraint constraint)
     {
-        JdbcTableHandle tableHandle = (JdbcTableHandle) table;
-        ConnectorTableLayout layout = new ConnectorTableLayout(new JdbcTableLayoutHandle(tableHandle, constraint.getSummary()));
-        return ImmutableList.of(new ConnectorTableLayoutResult(layout, constraint.getSummary()));
+        JdbcTableHandle jdbcHandle = (JdbcTableHandle) handle;
+        jdbcHandle = new JdbcTableHandle(
+                jdbcHandle.getSchemaTableName(),
+                jdbcHandle.getCatalogName(),
+                jdbcHandle.getSchemaName(),
+                jdbcHandle.getTableName(),
+                constraint.getSummary());
```

Comment:
The applyFilter method can be called multiple times throughout the optimization process. The connector needs to:
- remember the previously applied constraint and combine it with the new one. Otherwise, it can result in pushed-down constraints being "lost". For this connector, it won't cause correctness issues because the remaining filter is equal to the provided constraint, but it might result in seemingly non-deterministic pushdown depending on the sequence of calls the optimizer makes.
- return Optional.empty() if applying the constraint had no effect. Otherwise, the optimizer can loop forever.

Reply:
Fixed by using `TupleDomain.intersect()` and comparing the old and new results